spython/main/execute.py

```diff
@@ -21,7 +21,7 @@
 import os
 import sys
 
-def execute(self, image=None, command=None, app=None, writable=False, contain=False):
+def execute(self, image=None, command=None, app=None, writable=False, contain=False, bind=""):
```

Comment:
A few quick comments! I think that we would want this to be either a single string, or a list of binds. For example. as a user I should be able to specify any of the following:

```
[ '/host:/container', '/both' ]
'/both'
'/host:container'
None
```
And so it would make sense to have a default of None, and then to check if the argument is not a list (and make it a list if it isn't) and then for each entry in the list, prepare the string in the format `--bind %s` %value. And I would suggest to have a function to take any of these possible inputs and return a nice bind string. For the examples above we would get:

```
"--bind /host:/container --bind /both"
"--bind /both"
"--bind /host:container"
(function wouldn't be called for this case)
```
Does that make sense? And then we can have a test for each of these cases to ensure it returns the correct! If you want to give a shot at the above go for it, if not I'm happy to do it (I pulled your branch and could code this up first thing tomorrow).

Reply:
It is nice. How is this? https://github.com/singularityhub/singularity-cli/pull/7/commits/85d8e67f258cfb22938f3174af78e91b042e34b9